library/Vanilla/PermissionsTranslationTrait.php

```diff
@@ -100,6 +101,85 @@ private function consolidatePermissions(array $permissions) {
         return $result;
     }
 
+    /**
+     * Given a permissions object, return its data array in the explicit-scope format.
+     *
+     * @param \Vanilla\Permissions $permissions
+     * @return array
+     */
+    private function formatPermissions(Permissions $permissions) {
+        $raw = $permissions->getPermissions();
+        $global = [];
+        $categories = [];
+
+        /**
+         * Format an array of permission names.
+         *
+         * @param array $perms
+         * @return array
+         */
+        $format = function (array $perms) {
```

Comment:
I feel like this deserves to be pulled out into its own function

Reply:
I don't know that it'll have any real use outside of `formatPermissions` and with this being a trait, the class implementing it will get all the private methods. I'd rather limit the visibility by keeping it as an anonymous function, until we have to use it elsewhere.